tests/ui/editions/edition-keywords-2018-2015-parsing.rs

```diff
@@ -25,8 +25,6 @@ pub fn check_async() {
     if passes_tt!(r#async) == 1 {} // OK
     module::async(); //~ ERROR expected identifier, found keyword `async`
     module::r#async(); // OK
-
-    let _recovery_witness: () = 0; //~ ERROR mismatched types
```

Comment:
We can leave this in and change the comment to be `// not emitted because of the macro parsing error`.